hydra-node/test/Hydra/ModelSpec.hs

```diff
@@ -95,11 +95,14 @@ import Test.QuickCheck.DynamicLogic (
  )
 import Test.QuickCheck.Gen.Unsafe (Capture (Capture), capture)
 import Test.QuickCheck.Monadic (PropertyM, assert, monadic', monitor, run)
-import Test.QuickCheck.StateModel (Actions, RunModel, runActions, stateAfter, pattern Actions)
+import Test.QuickCheck.StateModel (Actions, RunModel, Step ((:=)), runActions, stateAfter, pattern Actions)
 import Test.Util (printTrace, traceInIOSim)
 
 spec :: Spec
 spec = do
+  -- There cannot be a UTxO with no ADAs
+  -- See https://github.com/input-output-hk/cardano-ledger/blob/master/doc/explanations/min-utxo-mary.rst
+  prop "model should not generate 0 Ada UTxO" $ withMaxSuccess 10000 prop_doesNotGenerate0AdaUTxO
```

Comment:
It feels a bit weird to check the model for that, but it's actually what you needed to fix - so it's fine I guess.

An alternative would have been to check the `genAdaValue` generator.

Reply:
we've decided to change genAdaValue for it to not generate 0 value Ada because we don't need them in our test model but, actually, it might be ok in some model, maybe later, for a 0 Ada value to be generated. What is not ok is for a UTxO to contain a 0 Ada value.
That's why we ended up testing the model per se for no 0 value UTxo. It opens for more ways of implementing this constraint.
But maybe the place of the test feels a bit odd 🤔